data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/buffer/BufferTopicConfig.java

```diff
@@ -102,6 +104,9 @@ class BufferTopicConfig extends CommonTopicConfig implements TopicProducerConfig
     @JsonProperty("fetch_max_bytes")
     private ByteCount fetchMaxBytes = DEFAULT_FETCH_MAX_BYTES;
 
+    @JsonProperty("max_message_bytes")
```

Comment:
Why do we need another property on topic when we have this on the producer properties?

It seems that if we set the producer properties value to any non-default, it makes sense to use that. Or perhaps if the producer properties is set to greater than the default, we change to the larger value.

Reply:
@dlvenable Kafka has separate properties for producer and topic. Producer properties are for client side property that is used when the KafkaProducer is created. Topic properties are used when the topic is created. We share producer properties with Kafka Sink. If we want to derive topic property for max_message_bytes when the producer properties is set, that's fine too. My current logic derives producer property when topic property is set.

Reply:
Ultimately this property matters only to writing to Kafka right? I think we should derive this value from the producer properties instead.

I'd probably say this is the general algorithm:

```
if(producerProperties.getMaxRequestSize() > DEFAULT_MAX_REQUEST_SIZE)
  setTopicMaxMessageBytes(producerProperties.getMaxRequestSize());
```

That is, if the client lowers the max size, we need not change the server. It is already compatible with the defaults. But, if the client increases to a size that would otherwise not work, then we do increase it.

Reply:
Were we going to remove this property in favor of using the ProducerProperties value?